### ISSUE TITLE
On MySQL JavaInstantColumnType  column uses DATETIME instead of TIMESTAMP

### ISSUE THREAD

**Issue by FRedEnergy** [JetBrains/Exposed#865]
Right now, when you have a column initialized using JavaInstantColumnType (or `timestamp()` from exposed-java-time module) it will use `DATETIME` type upon table creation via SchemaUtils if MySQL database is used.

I believe `TIMESTAMP` type from MySQL would be more suitable since it conveys the same meaning as Instant class in java (aka pure time in UTC), while `DATETIME` acts more like LocalDateTime. This behaviour can be kind of misleading and recently resulted in some unexpected issues on our production server (we naturally thought that timestamp() would produce an actual TIMESTAMP and never checked)

I realise that it's probably too late to change it but this detail of implementation at least should be addressed in documentation.  Some people could write their own implementations of JavaInstantColumnType which would use TIMESTAMP if they actualy need such behaviour.

**Comment by michaelcordero** [comment]
I think this duplicates my issue: #844

**Comment by FRedEnergy** [comment]
Not really, we're talking about the same column type but you propose kind of the opposite solution.

**Comment by michaelcordero** [comment]
Yes, if I understand you correctly, you want to change the existing API for JavaInstantColumnType to return `java.sql.timestamp` not `DATETIME`. While I am proposing an entirely new column type: `JavaTimestampColumnType` which would also return `java.sql.timestamp`, but not break backwards compatibility would it?

**Comment by bartvanheukelom** [comment]
I agree with OP that it would be good to have a column with JVM type `java.time.Instant` and (My)SQL type `TIMESTAMP`. Not `java.sql.Timestamp` and not SQL `DATETIME`.

I understand that changing the default for the `timestamp` function now is no longer possible or wise, but I do agree that its behaviour is unexpected and warrants more documentation.

Of course, it's pretty easy to create such a type oneself, but it's so common that it would be good to have in the library. Also, then it could be added to `DataTypeProvider` so `TIMESTAMP` would be used on e.g. MySQL, but whatever type is most appropriate in databases that don't support that.
